ts/Stock/StockToolsBindings.ts

```diff
@@ -499,51 +506,74 @@ extend(NavigationBindings.prototype, {
      * @param {number} defaultHeight
      *        Default height in percents.
      *
-     * @return {Array}
-     *         An array of calculated positions in percentages.
-     *         Format: `{top: Number, height: Number}`
+     * @param {string} removedYAxisHeight
+     *        Height of the removed yAxis.
+     *
+     * @return {Highcharts.YAxisPositions}
+     *         An object containing an array of calculated positions
+     *         in percentages. Format: `{top: Number, height: Number}`
+     *         and maximum value of top + height of axes.
      */
     getYAxisPositions: function (
         yAxes: Array<AxisType>,
         plotHeight: number,
-        defaultHeight: number
-    ): Array<Record<string, number>> {
-        var positions: (Array<Record<string, number>>|undefined),
-            allAxesHeight = 0;
-
+        defaultHeight: number,
+        removedYAxisHeight?: string
+    ): Highcharts.YAxisPositions {
+        var positions: Array<Record<string, number>>|undefined,
+            allAxesHeight = 0,
+            previousAxisHeight: number,
+            removedHeight: number;
         /** @private */
-        function isPercentage(prop: (number|string|undefined)): boolean {
+        function isPercentage(prop: number | string | undefined): boolean {
             return defined(prop) && !isNumber(prop) && (prop.match('%') as any);
         }
 
-        positions = yAxes.map(function (
-            yAxis: AxisType
-        ): Record<string, number> {
+        if (removedYAxisHeight) {
+            removedHeight = correctFloat((parseFloat(removedYAxisHeight) / 100));
+        }
+
+        positions = yAxes.map(function (yAxis: AxisType, index: number): Record<string, number> {
             var height = isPercentage(yAxis.options.height) ?
-                    parseFloat(yAxis.options.height as any) / 100 :
-                    yAxis.height / plotHeight,
+                    correctFloat(parseFloat(yAxis.options.height as any) / 100) :
+                    correctFloat(yAxis.height / plotHeight),
```

Comment:
Micro optimalization: instead of `isPercentage() ? correctFloat(a) : correctFloat(b)` you can use `correctFloat(isPercentage() ? a : b)`